### PR TITLE
I have added everything here

### PR DESCRIPTION
**Pull request by pallabi-ghosh**

### CODE REVIEW

File: databases/week2/Creating a database.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+CREATE DATABASE Cake_Bakery;
+
+USE Cake_Bakery;
+
+SET NAMES utf8mb4;
+
+  
+CREATE TABLE `bakery_offices` (
+  `offices_id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
```
Comment (babakbandpey) [inline databases/week2/Creating a database.sql]: simple `id` would do

File: databases/week2/Creating a database.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  `name` varchar(255) NOT NULL,
+  `description` varchar(255) NOT NULL,
+  `location` varchar(255) NOT NULL,
+  `created` datetime  NOT NULL
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `bakery_office_staffs` (
+`staff_id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
```
Comment (babakbandpey) [inline databases/week2/Creating a database.sql]: `id`

File: databases/week2/Creating a database.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+`name` varchar(255) NOT NULL,
+`age` int(10) unsigned NOT NULL,
+`position_id` int(10) unsigned NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/Creating a database.sql]: is this a foreign key?

File: databases/week2/Creating a database.sql [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+`position_title` varchar(255) NOT NULL,
+`offices_id` int(10) unsigned,
```
Comment (babakbandpey) [inline databases/week2/Creating a database.sql]: is this a foreign key?

File: databases/week2/Creating a database.sql [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+CONSTRAINT `fk_offices_id` FOREIGN KEY (`offices_id`) REFERENCES `bakery_offices` (`offices_id`) ON DELETE CASCADE ON UPDATE CASCADE
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+  
+  CREATE TABLE `baked_items` (
+`items_id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
```
Comment (babakbandpey) [inline databases/week2/Creating a database.sql]: `id`

File: databases/week2/Creating a database.sql [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+CREATE TABLE `bakery_office_staffs` (
+`staff_id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+`name` varchar(255) NOT NULL,
+`age` int(10) unsigned NOT NULL,
+`position_id` int(10) unsigned NOT NULL,
+`position_title` varchar(255) NOT NULL,
+`offices_id` int(10) unsigned,
+CONSTRAINT `fk_offices_id` FOREIGN KEY (`offices_id`) REFERENCES `bakery_offices` (`offices_id`) ON DELETE CASCADE ON UPDATE CASCADE
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
```
Comment (babakbandpey) [inline databases/week2/Creating a database.sql:36]: can a staff be working at two offices? and if so how would you change to accommodate it?

File: databases/week2/More queries.sql [hunk 7]
```diff
@@ -0,0 +1,31 @@
+SET NAMES utf8mb4;
+
+-- Get all tasks assigned to users whose email ends in @spotify.com --
+SELECT t.title
+FROM user_task ut 
+INNER JOIN task t ON ut.task_id=t.id 
+INNER JOIN (
+    SELECT u.id, u.email
```
Comment (babakbandpey) [inline databases/week2/More queries.sql]: When you only need the `u.id` why do you select the `u.email`?

File: databases/week2/More queries.sql [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    FROM user u
+    WHERE email LIKE '%@spotify.com'
+) u ON ut.user_id=u.id;
+
```
Comment (babakbandpey) [inline databases/week2/More queries.sql:46]: a simpler version:

```
SELECT 
    t.title
FROM
    user_task ut
        INNER JOIN
    task t ON ut.task_id = t.id
        INNER JOIN
    user u ON u.id = ut.user_id
        AND u.email LIKE '%spotify.com%'
```

File: databases/week2/More queries.sql [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+-- Get all taks for 'Donald Duck' where status 'not started' -- 
+SELECT DISTINCT user.name, task.id, task.title, task.description, status.name FROM user
```
Comment (babakbandpey) [inline databases/week2/More queries.sql]: Remove the `distinct`

File: databases/week2/More queries.sql [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+INNER JOIN user_task ON user.id = user_task.task_id
+INNER JOIN task ON task.id = user_task.task_id
+INNER JOIN status ON task.status_id = status_id
+WHERE user.name LIKE '%Donald Duck%' AND status.name = 'Not started';
+
+
```
Comment (babakbandpey) [inline databases/week2/More queries.sql:61]: if you use MySQL Workbench you can use CTRL+B to beautify your queries

```
SELECT 
    user.name,
    task.id,
    task.title,
    task.description,
    status.name
FROM
    user
        INNER JOIN
    user_task ON user.id = user_task.user_id
        INNER JOIN
    task ON task.id = user_task.task_id
        INNER JOIN
    status ON task.status_id = status_id
WHERE
    user.name LIKE '%Donald Duck%'
        AND status.name = 'Not started';
```

File: databases/week2/More queries.sql [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+-- Get all task for 'Maryrose Meadows' that were created in september --
+SELECT * FROM user JOIN user_task ON user.id = user_task.user_id JOIN task ON user_task.task_id = task.id WHERE user.name = 'Aubry Polak' AND MONTH(created) = 05;
+
```
Comment (babakbandpey) [inline databases/week2/More queries.sql]: Who is `Aubry Polak`?
`'Maryrose Meadows' `

You need to run your query before you submit it.

File: databases/week2/School database.sql [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+`ends` DATETIME NULL DEFAULT NULL
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+-- Student: with the columns: id, name, email, phone, class_id (foreign key) --
+CREATE TABLE `student`(
+`id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+`name` varchar(255) NOT NULL,
+`email` varchar(255) NOT NULL,
+`phone` varchar(255) NULL,
+`class_id` int(10) unsigned NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/School database.sql]: Remove this
`student_classes `is for this purpose

File: databases/week2/School database.sql [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+CONSTRAINT `fk_class` FOREIGN KEY (`class_id`) REFERENCES `class`(`id`) ON DELETE CASCADE ON UPDATE CASCADE
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+-- Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations). --
+CREATE TABLE `student_classes`(
```
Comment (babakbandpey) [inline databases/week2/School database.sql]: Missing a primary key

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello,
There are some changes you need to attend to
The alter table exercise is missing.

File: databases/week2/School database.sql (hunk 13 above)
Comment (pallabi-ghosh) [reply databases/week2/School database.sql]: But I have student_id defined as Primary Key on line 24. Maybe I am missing something here ?

File: databases/week2/Creating a database.sql (hunk 1 above)
Comment (pallabi-ghosh) [reply databases/week2/Creating a database.sql]: Is this only about the name :) ? Cant we just put anything as a column name ? In this case I have created it as 'offices_id' as I wanted to make sure that column name is understandable and not ambiguous with the other primary keys that we have in the rest of the tables.

File: databases/week2/Creating a database.sql (hunk 3 above)
Comment (pallabi-ghosh) [reply databases/week2/Creating a database.sql]: But , I do not have any constraints on this position_id. So it is not a foreign key. :)

File: databases/week2/Creating a database.sql (hunk 4 above)
Comment (pallabi-ghosh) [reply databases/week2/Creating a database.sql]: Yes

File: databases/week2/Creating a database.sql (hunk 1 above)
Comment (babakbandpey) [reply databases/week2/Creating a database.sql]: It is not just about the naming.
If you have fx a book, then it has a name. It has not a book_name, it has an id and not a book_id.
When you use this id as foreign key then you call it book_id, so when you see book_id you realize that it is a foreign key.
Another example is this:
if you are coding then you might create a class `Book `and an object of the class `book`
what do you prefer to write? `book.book_id` or `book.id`

File: databases/week2/Creating a database.sql (hunk 3 above)
Comment (babakbandpey) [reply databases/week2/Creating a database.sql]: What does `position_id` refer to? I am just wondering

Comment (babakbandpey) [reply databases/week2/Creating a database.sql]: You need a `positions` table if you like to do it elegantly

File: databases/week2/Creating a database.sql (hunk 6 above)
Comment (babakbandpey) [reply databases/week2/Creating a database.sql:36]: You need a `staffs` table

**Review by babakbandpey — CHANGES_REQUESTED** [review]
I still wish that you make the changes.
Thank you

**Review by babakbandpey — APPROVED** [review]
very good. Thank you.

**Pull request merged**